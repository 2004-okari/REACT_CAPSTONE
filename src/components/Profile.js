import React from 'react';
import { useSelector } from 'react-redux';

function Profile() {
  const rockets = useSelector((state) => state.rockets.rockets);
  const reservedRocket = rockets.filter((rocket) => rocket.reserved);

  return (
    <div>
      <section>
        <h3>My Rockets</h3>
        {reservedRocket.length === 0 ? <p>No rockets reserved</p> : (
          <ul>
            {reservedRocket.map((rocket) => (
              <li key={rocket.id}>{rocket.name}</li>
            ))}
          </ul>
        )}
      </section>
    </div>
  );
}

export default Profile;
