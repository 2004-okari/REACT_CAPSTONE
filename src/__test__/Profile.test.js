import React from 'react';
import renderer from 'react-test-renderer';
import Profile from '../components/Profile'; // Adjust the path
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureMockStore([]);

test('Profile component matches snapshot', () => {
  const store = mockStore({
    missions: {
      missions: [],
    },
    rockets: {
      rockets: [],
    },
  });

  const tree = renderer
    .create(
      <Provider store={store}>
        <Profile />
      </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
