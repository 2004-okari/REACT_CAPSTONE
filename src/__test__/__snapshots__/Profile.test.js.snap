// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component matches snapshot 1`] = `
<div
  className="profile-container"
>
  <section
    className="mission"
  >
    <h3>
      My Mission
    </h3>
    <p>
      No mission reserved
    </p>
  </section>
  <section>
    <h3>
      My Rockets
    </h3>
    <p>
      No rockets reserved
    </p>
  </section>
</div>
`;
